lib/fauna/client.rb

```diff
@@ -171,6 +172,8 @@ def execute(action, path, query = nil, data = nil)
           response_raw, response_content, response.status, response.headers,
           start_time, end_time)
 
+      @last_seen = response.headers['X-Txn-Time'] if response.headers.key?('X-Txn-Time')
```

Comment:
other drivers are testing if the returned `X-Txn-Time` is greater than the actual value, because two threads can run queries with the same driver and update that value with a timestamp in the past

Reply:
Yup. I just realized that while working on the C# implementation. Circling back to fix this. :)